content/k8s/fluxcd/helmrelease/_index.md

```diff
@@ -0,0 +1,178 @@
+---
+content_type: project
+flavours:
+- none
+from_repo: k8s/manual-app-deployment/project-overview
+prerequisites:
+  hard:
+  - k8s/manual-app-deployment/project-overview
+  soft:
+  - k8s/fluxcd/helmrepository
+ready: true
+submission_type: continue_repo
+tags:
+- kubernetes
+- fluxcd
+title: Helmrelease
+---
+
+
+ 
+
+## Helmrelease cert-manager
+
+To load a helmchart from a helmrepository using FluxCD's helmcontroller we
+issue a helmrelease.
+
+To start let's create folders and files for each release
+
+```
+|-- infrastructure
+|   |-- cert-manager
+|   |   |-- kustomization.yaml
+|   |   |-- kustomizeconfig.yaml
+|   |   |-- release.yaml
+|   |   |-- values.yaml
+|   |   |-- namespace.yaml
+|   |   |-- issuer.yaml
+```
+
+```
+# infrastructure/cert-manager/kustomization.yaml
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: cert-manager
+resources:
+  - namespace.yaml
+  - release.yaml
+configMapGenerator:
+  - name: values
+    files:
+      - values.yaml=values.yaml
+configurations:
+  - kustomizeconfig.yaml
+```
+
+```
+# infrastructure/cert-manager/kuztomizeconfig.yaml
+---
+# Kustomize config for enabling HelmRelease values from
+# ConfigMaps and Secrets generated by Kustomize
+nameReference:
+- kind: ConfigMap
+  version: v1
+  fieldSpecs:
+  - path: spec/valuesFrom/name
+    kind: HelmRelease
+```
+
+```
+# infrastructure/cert-manager/release.yaml
+---
+apiVersion: helm.toolkit.fluxcd.io/v2beta1
+kind: HelmRelease
+metadata:
+  name: cert-manager
+  namespace: cert-manager
+spec:
+  releaseName: cert-manager
+  chart:
+    spec:
+      chart: cert-manager
+      sourceRef:
+        kind: HelmRepository
+        name: jetstack
+        namespace: flux-system
+  interval: 24h0m0s
+  install:
+    remediation:
+      retries: 4
+  valuesFrom:
+    - kind: ConfigMap
+      name: values
+```
+
+```
+# infrastructure/cert-manager/namespace.yaml
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: cert-manager
+```
+
+```
+# infrastructure/cert-manager/issuer.yaml
+---
+apiVersion: cert-manager.io/v1
+kind: ClusterIssuer
+metadata:
+  name: letsencrypt-prod
+  namespace: cert-manager
+spec:
+  acme:
+    server: https://acme-v02.api.letsencrypt.org/directory
+    email: my@email.com
+    privateKeySecretRef:
+      name: letsencrypt-prod
+    solvers:
+      - http01:
+          ingress:
+            ingressClassName: nginx
+```
+
+Now remember to add the folder in the infrastructure kustomization
+
+```
+# infrastructure/kustomization.yaml
+---
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - sources
+  - cert-manager
+```
+
+git push to your main branch and you should notice the containers rollout in cert-manager namespace
+
+```
+kubectl -n cert-manager get pods
```

Comment:
It gives me an error:

```
infrastructure   45m   False   kustomize build failed: accumulating resources: accumulation err='accumulating resources from 'cert-manager': read /tmp/kustomization-4176172456/infrastructure/cert-manager: is a directory': recursed accumulation of path '/tmp/kustomization-4176172456/infrastructure/cert-manager': loading KV pairs: file sources: [values.yaml=values.yaml]: open /tmp/kustomization-4176172456/infrastructure/cert-manager/values.yaml: no such file or directory
```

It seems it doesn't work without a `values.yaml` file.

---

I just noticed that the first instruction (with the file tree) includes a `values.yaml` file. Might be better to state in the content to create an empty `values.yaml` file. Or leave to the student to figure it out 🤷